spec/buffered_line_reader_spec.rb

```diff
@@ -0,0 +1,62 @@
+require "ld-eventsource/impl/buffered_line_reader"
+
+def tests_for_terminator(term, desc)
+  def make_tests(name, input_line_chunks, expected_lines)
+    [{
+      name: "#{name}: one chunk per line",
+      chunks: input_line_chunks,
+      expected: expected_lines
+    }].concat(
+      (1..4).map do |size|
+        ({
+          name: "#{name}: #{size}-character chunks",
+          chunks: input_line_chunks.join().chars.to_a.each_slice(size).map { |a| a.join },
+          expected: expected_lines
+        })
+      end
+    )
+  end
+  [
+    make_tests("non-empty lines",
+      ["first line" + term, "second line" + term, "3rd line" + term],
+      ["first line", "second line", "3rd line"]),
+
+    make_tests("empty first line",
+      [term, "second line" + term, "3rd line" + term],
+      ["", "second line", "3rd line"]),
+
+    make_tests("empty middle line",
+      ["first line" + term, term, "3rd line" + term],
+      ["first line", "", "3rd line"]),
+
+    make_tests("series of empty lines",
+      ["first line" + term, term, term, term, "3rd line" + term],
+      ["first line", "", "", "", "3rd line"]),
+
+    make_tests("multi-line chunks",
+      ["first line" + term + "second line" + term + "third",
+       " line" + term + "fourth line" + term],
+      ["first line", "second line", "third line", "fourth line"])
+  ].flatten
+end
+
+describe SSE::Impl::BufferedLineReader do
+  subject { SSE::Impl::BufferedLineReader }
+
+  terminators = {
```

Comment:
Is it possible for SSE to have multiple kinds of terminators in a message? Even so, the tests that are here maybe enough to verify that case work work already even if that is possible.

Reply:
It's unlikely that any server is implemented that way, but the spec does allow it, so it's reasonable to add some tests for that.